export_snowflake/__init__.py

```diff
@@ -521,6 +519,134 @@ def flush_records(stream: str,
+    stream = config["stream"]
+    LOGGER.info(f"Elapsed time usage for {stream} is {parallel_transfer_end_time - parallel_transfer_start_time}")
+
+
+def export_to_snowflake(config, o, file_format_type): 
+    # retrieve the csv.gz lists and the schema file from the s3 bucket
+    s3_client = boto3.client('s3')
+
+    bucket = config["bucket"]
+    prefix = config["prefix"]
+    response = s3_client.list_objects_v2(Bucket=bucket, Prefix=prefix)
+
+    s3_csv_lists = []
+    schema_file = None
+    for obj in response['Contents']:
+        if obj['Key'].endswith('.csv.gz'):
+            # append the exportTaskID to the file name to avoid conflict
+            file_name = obj['Key'].split("/")[-2] + '-' + obj['Key'].split("/")[-1]
+            s3_csv_lists.append((bucket, obj['Key'], file_name, config["stream"]))
+        # for both python and spark parquet_to_csv steps, we always have a schema.json with the schema details
+        if obj['Key'].endswith('.json'):
+            schema_file = obj['Key']
+
+    if schema_file is None:
+        raise Exception("Schema file not found in the s3 bucket")
+    if len(s3_csv_lists) == 0:
+        raise Exception("No csv files found in the s3 bucket")
+
+    local_schema_file = 'local_schema.json'
+    s3_client.download_file(bucket, schema_file, local_schema_file)
+
+    f = open(local_schema_file)
+    schema = json.load(f)
+
+    # generate the schema object for DbSync
+    stream = config["stream"]
+    o = {
+        "stream": stream,
+        "schema": {"properties": schema['properties']},
+        "key_properties": config["key_columns"]
+    }
+
+    # check if the schema(Snowflake table) exists and create it if not
+    # sync_table will check the schema with the table in Snowflake, if there's a miss matching in columns, raise an error
+    db_sync = DbSync(config, o, None, file_format_type)
+    db_sync.create_schema_if_not_exists()
+    db_sync.sync_table()
+
+    # the csv files in s3 are compressed, changed the parquet_to_csv to write into gzip format, also remove the headers
+    transfer_data_from_s3_to_snowflake(s3_csv_lists, config, o, file_format_type)
+
+    # Snowflake locks the table when we run query against it, so we need to load the file sequentially
+    try: 
+        start_time = time.time()
+        for obj in s3_csv_lists:
+            stage_file_path = obj[2].replace("\\", "/")
+            
+            db_sync.load_file(stage_file_path)
+        end_time = time.time()
+        LOGGER.info(f"Elapsed time usage to load {len(s3_csv_lists)} files for {stream}: {end_time - start_time}")
+    except Exception as e:
+        LOGGER.info("Error occurred while loading the file: ", e)
+        # errors are already handled inside the load_file function, just raise here
+        raise e
+    finally:
+        # delete the remote file anyway
+        # Snowflake will charge for the memory usage if we don't delete the file
+        for obj in s3_csv_lists: 
```

Comment:
we don't want to fail if cleanup tasks fails.

Reply:
after the change to process from s3 to snowflake directly, no need to clean up the stage files. Just have to check if the stage exists and delete if so.